src/Data/Yaml.hs

```diff
@@ -100,65 +106,89 @@ import qualified Data.Text.Lazy as TL
 import Data.Text.Lazy.Builder (toLazyText)
 import qualified Data.Vector as V
 import System.IO.Unsafe (unsafePerformIO)
+import Data.Text (Text)
 
 import Data.Yaml.Internal
-import Text.Libyaml hiding (encode, decode, encodeFile, decodeFile)
+import Text.Libyaml hiding (encode, decode, encodeFile, decodeFile, encodeWith, encodeFileWith)
 import qualified Text.Libyaml as Y
 
+data EncodeOptions = EncodeOptions
+  { encodeOptionsStringStyle :: Text -> ( Tag, Style )
+  , encodeOptionsFormat :: FormatOptions
+  }
+
+-- | Determine whether a string must be quoted in YAML and can't appear as plain text.
+-- Useful if you want to write a custom 'encodeOptionsStringStyle'.
+isSpecialString :: Text -> Bool
+isSpecialString s = s `HashSet.member` specialStrings || isNumeric s
+
+defaultEncodeOptions :: EncodeOptions
+defaultEncodeOptions = EncodeOptions
+  { encodeOptionsStringStyle = \s ->
+    -- Empty strings need special handling to ensure they get quoted. This avoids:
+    -- https://github.com/snoyberg/yaml/issues/24
+    if isSpecialString s
+      then ( NoTag, SingleQuoted )
+      else ( StrTag, PlainNoTag )
+  , encodeOptionsFormat = defaultFormatOptions
+  }
+
 -- | Encode a value into its YAML representation.
 encode :: ToJSON a => a -> ByteString
-encode obj = unsafePerformIO $ runConduitRes
-    $ CL.sourceList (objToEvents $ toJSON obj)
-   .| Y.encode
+encode = encodeWith defaultEncodeOptions
+
+-- | Encode a value into its YAML representation with custom styling.
```

Comment:
Please add a `@since` comment with the newest version number, and include a `ChangeLog` entry and cabal file bump for this new version.